docs/porting.md

```diff
@@ -156,9 +156,49 @@ after importing a module that does not support the GIL.
     free-threaded builds. See [the docs on setting up CI](ci.md) for advice on
     how to build projects that depend on Cython.
 
+=== "Rust"
+    If you use the CPython C API via [PyO3](https://pyo3.rs), then you
+    can follow the [PyO3 Guide
+    section](https://pyo3.rs/v0.23.1/free-threading.html) on supporting
+    free-threaded Python. You must also update your extension to at least
+    version 0.23.
+
+    You should write multithreaded tests of any code you expose to Python. You
+    can leverage your existing test suite and
+    [`pytest-run-parallel`](https://github.com/Quansight-Labs/pytest-run-parallel)
+    to discover thread safety issues due to use of global state, but you should
+    also write explicitly multithreaded tests using any data structures provided
+    by modules defined in Rust extensions.
```

Comment:
It would be good to explain why this is necessary (more so in Rust than for C/C++ extensions it sounds like?), to motivate the reader to actually do it.

Reply:
IMO we should probably be recommending that for C/C++ if we aren't already. I'll just say to see the recommendations for testing C/C++ libraries.

Reply:
I did this and also fleshed out the general suggestions for testing and moved some cython-specific stuff to the cython tab.